### ISSUE TITLE
Validation on CPU (with csv generator)

### ISSUE THREAD

**Issue by alexattia** [fizyr/keras-retinanet#448]
Hello,   
First of all thank you very much for this great repo.   
I am training a model (ResNet 50 backbone) using multiple GPUs with CSV annotations files. The training is done correctly.   
However, I am also validating my model (at the end of each epoch) with a CSV annotation file but the validation is computed on CPU mostly (a tiny part is done on one GPU, checked with Chrome Tracing and `nvidia-smi`). It occurs during `predict_on_batch` in the evaluation part, I think that the prediction model is not converted to multiple GPUs but I don't understand why it's on CPU and not even one GPU.   
Thank you very much,

**Comment by hgaiser** [comment]
Hey,
Yes, with multi-gpu the model is created on the CPU to reduce the amount of memory needed. Keras then wraps this model, dividing the load over different GPUs. This "divided" model can't be used for prediction however and there is (likely) no room in memory for loading a copy of the network. The result is that the evaluation code uses the CPU generated model. Perhaps we should check for multi-gpu and disable evaluation in that case.

tldr; Keras' multi-gpu training is problematic, forcing us to use the CPU to do evaluation. Personally, I train without evaluation enabled and manually run the evaluation tool on a separate machine from time to time.

As this unfortunately can't be fixed with Keras' implementation, I'm going to close this issue.
